### PR TITLE
Ignore edits to issues predating the bot

### PR DESCRIPTION
**Pull request by strugee**

### CODE REVIEW

**Comment by strugee** [comment]
Should fix the scenario described in https://github.com/EFForg/https-everywhere/issues/10843#issuecomment-340945274.

File: utils/issue-format-bot/lib/issueedit.js [hunk 1]
```diff
@@ -20,6 +20,13 @@ module.exports = function(robot, alexa) {
       return;
     }
 
+    const createdAt = new Date(context.payload.issue.created_at);
+    // September 25th, 2017
+    if (createdAt.getFullYear() <= 2017 && createdAt.getMonth() <= 8 && createdAt.getDate() <= 25) {
```
Comment (Hainish) [inline utils/issue-format-bot/lib/issueedit.js]: This conditional will fail for an issue created in September, 2016.

Comment (Hainish) [reply utils/issue-format-bot/lib/issueedit.js]: The cleanest way to do this without a long conditional is probably formatted string comparison.  Munge `createdAt` into `YYYY-MM-DD` format, then compare it to `"2017-08-25"`

Comment (koops76) [reply utils/issue-format-bot/lib/issueedit.js]: Can't we compare `Date`s directly, or at least by conversion to timestamps?

Comment (koops76) [inline utils/issue-format-bot/lib/issueedit.js]:
```js
const botCreatedDate = new Date(2017, 10, 25)
if (createdAt < botCreatedDate) {
```

Comment (koops76) [reply utils/issue-format-bot/lib/issueedit.js]: @Hainish Better way is to compare dates directly.

Comment (strugee) [reply utils/issue-format-bot/lib/issueedit.js]: @Hainish nice catch. Will push a fix based on https://github.com/EFForg/https-everywhere/pull/13460#discussion_r148623578

Comment (strugee) [reply utils/issue-format-bot/lib/issueedit.js]: Nice, thanks! I had no idea you could compare `Date`s like this; I was gonna compare Unix timestamps.

**Comment by strugee** [comment]
Fixed the date comparison and rebased on master.

**Comment by strugee** [comment]
Also pushed an unrelated test suite improvement that didn't seem worth sending a separate PR for.

**Pull request merged**